File: simple-statemachine-with-persistence/uising-statemachine-data-jpa/simple-statemachine-java/src/main/java/com/example/statemachine/simple/doument/config/StateMachineServiceConfig.java
package com.example.statemachine.simple.doument.config;

import com.example.statemachine.simple.doument.DocumentEvent;
import com.example.statemachine.simple.doument.DocumentState;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.statemachine.config.StateMachineFactory;
import org.springframework.statemachine.persist.StateMachineRuntimePersister;
import org.springframework.statemachine.service.DefaultStateMachineService;
import org.springframework.statemachine.service.StateMachineService;

@Configuration
public class StateMachineServiceConfig {

  @Bean
  public StateMachineService<DocumentState, DocumentEvent> stateMachineService(
    StateMachineFactory<DocumentState, DocumentEvent> stateMachineFactory,
    StateMachineRuntimePersister<DocumentState, DocumentEvent, String> stateMachineRuntimePersister) {

    return new DefaultStateMachineService<>(stateMachineFactory, stateMachineRuntimePersister);
  }

}

File: simple-statemachine-with-persistence/uising-statemachine-data-jpa/simple-statemachine-kotlin/src/main/kotlin/com/example/statemachine/simple/document/config/StateMachineServiceConfig.kt
package com.example.statemachine.simple.document.config

import com.example.statemachine.simple.document.DocumentEvent
import com.example.statemachine.simple.document.DocumentState
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.statemachine.config.StateMachineFactory
import org.springframework.statemachine.persist.StateMachineRuntimePersister
import org.springframework.statemachine.service.DefaultStateMachineService
import org.springframework.statemachine.service.StateMachineService

@Configuration
class StateMachineServiceConfig {
  @Bean
  fun stateMachineService(
    stateMachineFactory: StateMachineFactory<DocumentState, DocumentEvent>?,
    stateMachineRuntimePersister: StateMachineRuntimePersister<DocumentState, DocumentEvent, String?>?
  ): StateMachineService<DocumentState, DocumentEvent> {
    return DefaultStateMachineService(stateMachineFactory, stateMachineRuntimePersister)
  }
}
